aws/resource_aws_lb_listener.go

```diff
@@ -252,15 +252,15 @@ func resourceAwsAlbListenerDelete(d *schema.ResourceData, meta interface{}) erro
 	return nil
 }
 
-func validateAwsAlbListenerPort(v interface{}, k string) (ws []string, errors []error) {
+func validateAwsLbListenerPort(v interface{}, k string) (ws []string, errors []error) {
 	port := v.(int)
 	if port < 1 || port > 65536 {
 		errors = append(errors, fmt.Errorf("%q must be a valid port number (1-65536)", k))
 	}
 	return
 }
 
-func validateAwsAlbListenerProtocol(v interface{}, k string) (ws []string, errors []error) {
+func validateAwsLbListenerProtocol(v interface{}, k string) (ws []string, errors []error) {
 	value := strings.ToLower(v.(string))
 	if value == "http" || value == "https" {
```

Comment:
Network Load Balancers only support TCP listeners, so I think line 265 needs to be updated to allow for a value of "tcp".

https://github.com/aws/aws-sdk-go/blob/master/service/elbv2/api.go#L3294-L3296

Reply:
@jwinter you are indeed correct - just pushed this change!